regionurlprovider.go

```diff
@@ -89,54 +129,9 @@ func (r *regionURLProvider) ReportAttemptFailure(cloudHostname, regionURL string
 	}
 
 	// remove failed region from regionSettings
-	for i, region := range hostnameSettings.regionSettings.Regions {
-		if region.Url == regionURL {
-			hostnameSettings.regionSettings.Regions = append(
-				hostnameSettings.regionSettings.Regions[:i],
-				hostnameSettings.regionSettings.Regions[i+1:]...,
-			)
-			break
-		}
-	}
-
-	return nil
-}
-
-// assume this is being called within a lock
-func (r *regionURLProvider) refreshRegionSettings(cloudHostname, token string) error {
-	settingsURL := "https://" + cloudHostname + "/settings/regions"
-	req, err := http.NewRequest("GET", settingsURL, nil)
-	if err != nil {
-		return errors.New("refreshRegionSettings failed to create request: " + err.Error())
-	}
-	req.Header = http.Header{
-		"Authorization": []string{"Bearer " + token},
-	}
-
-	resp, err := r.httpClient.Do(req)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
-
-	if resp.StatusCode != http.StatusOK {
-		return errors.New("refreshRegionSettings failed to fetch region settings. http status: " + resp.Status)
-	}
-
-	respBody, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return errors.New("refreshRegionSettings failed to read response body: " + err.Error())
-	}
-	regions := &livekit.RegionSettings{}
-	if err := protojson.Unmarshal(respBody, regions); err != nil {
-		return errors.New("refreshRegionSettings failed to decode region settings: " + err.Error())
-	}
-
-	r.hostnameSettingsCache[cloudHostname] = &hostnameSettingsCacheItem{
-		regionSettings:    regions,
-		updatedAt:         time.Now(),
-		regionURLAttempts: map[string]int{},
-	}
+	_ = slices.DeleteFunc(hostnameSettings.regionSettings.Regions, func(region *livekit.RegionInfo) bool {
```

Comment:
the result needs to be assigned back to `hostnameSettings.regionSettings.Regions` - it works like `append`